public/js/logic.js

```diff
@@ -0,0 +1,74 @@
+function fetchNode(url, data) {
+
+    return fetch(url, {
+        method: "POST",
+        body: JSON.stringify(data),
```

Comment:
this will always use POST method, but your use case above is to retrieve data, so it is more convenient to use GET and you can send your search term along with querystring instead of the request body.

```js
fetch('/search?q=${input.value}')
```

[GET vs POST](https://stackoverflow.com/questions/3477333/what-is-the-difference-between-post-and-get)